### ISSUE TITLE
Unable to set NET_EEPROM_OFFSET from sketch

### ISSUE THREAD

**Issue by per1234** [gregington/NetEEPROM#2]
From https://github.com/gregington/NetEEPROM/blob/master/src/NetEEPROM.h#L10-L11:
```
 * Sketches can override this by defining NET_EEPROM_OFFSET before
 * importing NetEEPROM.h.
```
This won't work because NetEEPROM.cpp can't see the define in the sketch so the value will always be set to 0. I do think it's useful to allow the offset to be set in the sketch rather than requiring the user to edit the library source and I can think of a few solutions to allow this:

1. Move all the function definitions that use NET_EEPROM_OFFSET to NetEEPROM.h where the define in the sketch is seen. Since this is most of the functions it might be best just to move everything to NetEEPROM.h. This has the advantage of not requiring an API change but it's annoying to not be able to use a .cpp file and I try to avoid use of the preprocessor unless absolutely necessary.

2. Add a setEEPROMoffset() function.

3. Set the value as an argument to the constructor. Probably not a good option since it's a breaking change.

**Comment by gregington** [comment]
Agreed that this is as issue.
I also agree that option 2 is the best one to maintain backwards compatibility.

**Comment by gregington** [comment]
Fixed in version 1.1.0.
